playbooks/migrating-postgres-with-rubyrep.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Migrating Postgres with Rubyrep
+description: How to migrate a Postgres to a new instance using Rubyrep
+---
+
+# Migrating Postgres with Rubyrep
+
+To set up database replication via [rubyrep](rubyrep.org), do the following:
+
+1) Create the new Postgres database and user
+
+2) Start a shell in the application with the new database connection and load the app schema with:
+```
+hokusai [staging|production] run --env "DATABASE_URL=postgres://{new database credentials}" "bundle exec rake db:schema:load"
+```
+
+3) Check the database schemas.
+  3a) If you want to download a dump of the current db to a local postgres, you can check schemas by running `rake db:schema:dump` locally, and checking in the diff in `db/schema.rb`.  You can also inspect the `schema_migrations` tables across the databases and compare to migrations.
+  3b) Log into the new database and note all foreign key constraints.  You can get foreign key constraints for each table with `\d+ table`.  Save the constraints to later re-enable.
```

Comment:
One-liner to list all foreign key constraints for all tables in the `public` schema...
```
SELECT conrelid::regclass AS table_from
     , conname
     , pg_get_constraintdef(oid)
FROM   pg_constraint
WHERE  contype IN ('f')
AND    connamespace = 'public'::regnamespace  -- your schema here
ORDER  BY conrelid::regclass::text, contype DESC;
```